### PR TITLE
Mes-6865: routing refactor

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Description

refactor routing and consolidate pages

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/non-pass-finalisation/non-pass-finalisation.page.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ import { TestFlowPageNames } from '@pages/page-names.constants';
 
 @Component({
   selector: 'app-non-pass-finalisation-cat-b',
```
Comment (matthew2564) [inline src/app/pages/non-pass-finalisation/non-pass-finalisation.page.ts]: remove category from selector

Comment (andrewsetterfield) [reply src/app/pages/non-pass-finalisation/non-pass-finalisation.page.ts]: good spot, hopefully there arnt more, i'll check

Comment (andrewsetterfield) [reply src/app/pages/non-pass-finalisation/non-pass-finalisation.page.ts]: rekey upload outcome has same issue

File: src/app/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export class DebriefCatAMod1Page implements OnInit {
   }
 
   async navigatePostDebriefHolding(): Promise<void> {
-    await this.routeByCat.navigateToPage(TestFlowPageNames.POST_DEBRIEF_HOLDING_PAGE, TestCategory.EUAM1);
+    await this.routeByCat.navigateToPage(TestFlowPageNames.POST_DEBRIEF_HOLDING_PAGE);
```
Comment (tomcrawleyy) [inline src/app/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts:31]: is routeByCategoryProvider the right one to be using here? or should it be router.navigate([POST_DEBRIEF_HOLDING_PAGE])?

File: src/app/pages/health-declaration/__tests__/health-declaration.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+  let component: HealthDeclarationPage;
+  let fixture: ComponentFixture<HealthDeclarationPage>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [RekeyReasonCatBEPage],
+      declarations: [HealthDeclarationPage],
       imports: [IonicModule.forRoot()],
       providers: [
         { provide: NavController, useClass: NavMock },
```
Comment (tomcrawleyy) [inline src/app/pages/health-declaration/__tests__/health-declaration.page.spec.ts:18]: whats navcontroller being provided/mocked for? most navcontroller stuff replaced by router

File: src/app/pages/non-pass-finalisation/__tests__/non-pass-finalisation.page.spec.ts [hunk 4]
```diff
@@ -4,23 +4,23 @@ import { NavMock } from '@mocks/angular-mocks/nav-mock';
...
+  let component: NonPassFinalisationPage;
+  let fixture: ComponentFixture<NonPassFinalisationPage>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [BackToOfficeCatCPCPage],
+      declarations: [NonPassFinalisationPage],
       imports: [IonicModule.forRoot()],
       providers: [
         { provide: NavController, useClass: NavMock },
```
Comment (tomcrawleyy) [inline src/app/pages/non-pass-finalisation/__tests__/non-pass-finalisation.page.spec.ts:18]: same here

File: src/app/pages/rekey-reason/__tests__/rekey-reason.page.spec.ts [hunk 5]
```diff
@@ -4,23 +4,23 @@ import { NavMock } from '@mocks/angular-mocks/nav-mock';
...
+  let component: RekeyReasonPage;
+  let fixture: ComponentFixture<RekeyReasonPage>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [RekeyReasonCatBPage],
+      declarations: [RekeyReasonPage],
       imports: [IonicModule.forRoot()],
       providers: [
         { provide: NavController, useClass: NavMock },
```
Comment (tomcrawleyy) [inline src/app/pages/rekey-reason/__tests__/rekey-reason.page.spec.ts:18]: and here

File: src/components/test-slot/test-outcome/test-outcome.ts [hunk 6]
```diff
@@ -474,45 +474,45 @@ export class TestOutcomeComponent implements OnInit {
...
+  //     case TestCategory.DE:
+  //       return [CAT_D.NON_PASS_FINALISATION_PAGE];
+  //     case TestCategory.K:
+  //     case TestCategory.H:
+  //     case TestCategory.G:
+  //     case TestCategory.F:
+  //       return [CAT_HOME_TEST.NON_PASS_FINALISATION_PAGE];
+  //     default:
+  //   }
+  // }
```
Comment (tomcrawleyy) [inline src/components/test-slot/test-outcome/test-outcome.ts]: should this be commented out?

File: src/app/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts [hunk 7]
```diff
@@ -33,6 +32,6 @@ export class WaitingRoomCatAdiPart2Page implements OnInit {
   }
 
   async navigateForward(): Promise<void> {
-    await this.routeByCat.navigateToPage(TestFlowPageNames.COMMUNICATION_PAGE, TestCategory.ADI2);
+    await this.routeByCat.navigateToPage(TestFlowPageNames.COMMUNICATION_PAGE);
```
Comment (tomcrawleyy) [inline src/app/pages/waiting-room/cat-adi-part2/waiting-room.cat-adi-part2.page.ts:35]: this routeByCategoryProvider is used everywhere across pr where typically would be router

File: src/app/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts (hunk 2 above)
Comment (andrewsetterfield) [reply src/app/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts:31]: @tomcrawleyy as the navigateToPage function simply calls router.navigate if u dont call it with a category, your right we could just call router.navigate directly.  I left it as this function so all navigation was consistant

File: src/components/test-slot/test-outcome/test-outcome.ts (hunk 6 above)
Comment (andrewsetterfield) [reply src/components/test-slot/test-outcome/test-outcome.ts]: its redundent now as non pass finalisation page isnt category specific. have removed

**Review by tomcrawleyy — APPROVED** [review]

**Review by Cullima — APPROVED** [review]

**Pull request merged**